### PR TITLE
fix endpoint

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

**Comment by StrongMonkey** [comment]
https://github.com/rancher/rancher/issues/7556

**Comment by alena1108** [comment]
@StrongMonkey can you reenable the previously failing test?

File: code/iaas/model/src/main/java/io/cattle/platform/core/dao/impl/InstanceDaoImpl.java [hunk 1]
```diff
@@ -319,10 +322,23 @@ protected PublicEndpoint map(List<Object> input) {
 
                 String address = "";
                 IpAddress ip = (IpAddress) input.get(3);
-                if (ip != null) {
+                if (ip != null && !ip.getAddress().equals("0.0.0.0")) {
                     address = ip.getAddress();
                 } else {
                     address = DataAccessor.fieldString(port, PortConstants.FIELD_BIND_ADDR);
+                    if (address.equals("0.0.0.0")){
```
Comment (cjellick) [inline code/iaas/model/src/main/java/io/cattle/platform/core/dao/impl/InstanceDaoImpl.java]: protect against null pointer

**Comment by StrongMonkey** [comment]
@alena1108 @cjellick i ll close this pr in favor of https://github.com/rancher/cattle/pull/2446

**Pull request closed without merging**